### PR TITLE
chore: ensure actions are pinned by commit hash

### PR DESCRIPTION
**Pull request by aquint-zama**
closes: _please link all relevant issues_

### PR content/description

Add a check to verify all actions used in CI workflow are pinned by commit hash and not by tag for security reason.

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

File: .github/workflows/ci_lint.yml [hunk 1]
```diff
@@ -25,3 +25,9 @@ jobs:
       - name: Lint workflows
         run: |
           make lint_workflow
+
+      - name: Ensure SHA pinned actions
+        uses: zgosalvez/github-actions-ensure-sha-pinned-actions@0901cf7b71c7ea6261ec69a3dc2bd3f9264f893e # v3.0.12
+        with:
+          allowlist: |
+            slsa-framework/slsa-github-generator
```
Comment (IceTDrinker) [inline .github/workflows/ci_lint.yml:33]: can't we use a commit hash for this one ?

Comment (aquint-zama) [reply .github/workflows/ci_lint.yml:33]: not yet unfortunately https://github.com/slsa-framework/slsa-github-generator/blob/main/README.md#referencing-slsa-builders-and-generators

Comment (IceTDrinker) [reply .github/workflows/ci_lint.yml:33]: ok

**Review by IceTDrinker — APPROVED** [review]
Thanks !

**Comment by IceTDrinker** [comment]
go for merge

**Pull request merged**